src/rest/handlers/rest_edges.erl

```diff
@@ -185,120 +185,78 @@ modify_state(delete, From, To, _, _, _) ->
-                     catch Type:Reason:Stacktrace ->
-                {'$pmap_error', self(), Type, Reason, Stacktrace}
-                     end,
-            Parent ! {Ref, self(), Result}
-              end)
-                     end, Edges),
+            Result =
+                try From = gmm_utils:create_vertex_id(SrcZone,  FromName),
+                To = gmm_utils:create_vertex_id(DstZone, ToName),
+                Vertex = case Successive of false -> From; true -> To end,
+                case [graph:vertex_exists(Vertex), graph:edge_exists(From, To)] of
+                    [{ok, true}, {ok, false}] -> true;
+                    _ -> false
+                end
+                catch Type:Reason:Stacktrace -> {'$pmap_error', self(), Type, Reason, Stacktrace} end,
+            Parent ! {self(), Result} end) end, Edges),
 
     % GATHERING RESULTS
-    Gather = fun
-    %PidsOrResults is initially the list of pids, gradually replaced by corresponding results
-                 F(PendingPids = [_ | _], PidsOrResults) ->
-                     receive
-                         {Ref, Pid, Result} ->
-                             NewPidsOrResults = replace(Pid, Result, PidsOrResults),
-                             F(lists:delete(Pid, PendingPids), NewPidsOrResults)
-                     after 5000 ->
-                         case lists:any(fun erlang:is_process_alive/1, PendingPids) of
-                             true ->
-                                 F(PendingPids, PidsOrResults);
-                             false ->
-                                 error({parallel_call_failed, {processes_dead, Pids}})
-                         end
-                     end;
-                 % wait for all pids to report back and then look for errors
-                 F([], AllResults) ->
-                     Errors = lists:filtermap(fun
-                                                  ({'$pmap_error', Pid, Type, Reason, Stacktrace}) ->
-                                                      {true, {Pid, Type, Reason, Stacktrace}};
-                                                  (_) ->
-                                                      false
-                                              end, AllResults),
-
-                     case Errors of
-                         [] ->
-                             lists:all(fun(X) -> X end, AllResults);
-                         _ ->
-                             false
-                     end
-             end,
+    Gather = fun F(PendingPids = [_ | _], PidsOrResults) ->
+        receive {Pid, Result} ->
+            NewPidsOrResults = rest_utils:replace(Pid, Result, PidsOrResults),
+            F(lists:delete(Pid, PendingPids), NewPidsOrResults)
+        after 5000 ->
+            case lists:any(fun erlang:is_process_alive/1, PendingPids) of
+                true -> F(PendingPids, PidsOrResults);
+                false -> error({parallel_call_failed, {processes_dead, Pids}}) end
+            end;
```

Comment:
this section should be indented one tab more, as its body of one branch of this lambda-function